src/Game/World.cs

```diff
@@ -576,304 +577,158 @@ GraphicEffectBlendMode blendmode
             );
         }
 
-        public static uint SearchObject(uint serial, ScanTypeObject scanType, ScanModeObject scanMode)
+        public static uint FindNearest(ScanTypeObject scanType)
```

Comment:
All of the changes here just avoid doing multiple iterations over the set of items/mobiles as it is searching.